### PR TITLE
devtools: Honor `acceptInsecureCerts` and `ignoreHTTPSErrors` settings

### LINKED ISSUE #10328: [🐛 Bug]: acceptInsecureCerts capability not being respected with devtools automationProtocol
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md#reporting-new-issues).

### WebdriverIO Version

latest

### Node.js Version

18.16.0

### Mode

WDIO Testrunner

### Which capabilities are you using?

```typescript
[
{
"browserName": "chrome"
},
{
"browserName": "firefox"
}
]
```

### What happened?

Specifying the capability `acceptInsecureCerts` to true does not enable the browsers to ignore self-signed certificates (often user with local web development servers).

Testing against local web development servers often require automation to ignore self-signed certificates. There is a [W3C spec capability](https://www.w3.org/TR/webdriver/#capabilities) (`acceptInsecureCerts`) that allows us to proceed in these cases. However the automation protocol `devtools` was not respecting this capability and passing the need flags to the browser launcher routines.

We already have the devtools option `ignoreHTTPSErrors` that can be defined in the `wdio:devtoolsOptions` capability. This seems to only have effect for Firefox.

For Webkit/Blink based browsers, it is necessary to pass the (chrome)flag `--ignore-certificate-errors`

An example of a domain with a "bad"  certificate could be found here: https://self-signed.badssl.com/

### What is your expected behavior?

Enabling the capability `acceptInsecureCerts` to true should allow me to reach and test domains that have a self-signed certificate.

### How to reproduce the bug.

with config:
```ts
{
automationProtocol: 'devtools',
capabilities: [
  {
    "browserName": "chrome",
    "acceptInsecureCerts": true
  },
  {
    "browserName": "firefox",
    "acceptInsecureCerts": true
  }
]
}
```

try to reach and assert something in a web page with a self-signed certificate:

```ts
await browser.url('https://self-signed.badssl.com/')
```

### Relevant log output

```typescript
N\A
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [X] I have searched the existing issues

### PR DESCRIPTION
**Pull request by nextlevelbeard**
## Proposed changes

Testing against local web development servers often require automation to ignore self-signed certificates. There is a W3C spec capability (`acceptInsecureCerts`) that allows us to proceed in these cases. However the automation protocol `devtools` was not respecting this capability and passing the need flags to the browser launcher routines.

We already have the devtools option `ignoreHTTPSErrors` that can be defined in the `wdio:devtoolsOptions` capability. This seems to only have effect for Firefox.

For Webkit/Blink based browsers, it is necessary to pass the (chrome)flag `--ignore-certificate-errors`.

There is some code duplication due to separated logic of launching Chrome and Edge. 

fixes #10328

[//]: # (Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.)

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [X] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md) doc
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have added necessary documentation (if appropriate)
- [x] I have added proper type definitions for new commands (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

**Comment by nextlevelbeard** [comment]
@christian-bromann Maybe we should cleanup this logic or at least align implementation.

`ignoreHTTPSErrors` is supposed to ignore more than just bad certificates, I suppose, but it's only really meant to be used with Firefox in Puppeteer, AFAIK.

For this use case we are going to have two settings that achieve the same thing: `ignoreHTTPSErrors` and W3C spec `acceptInsecureCerts`.

Can you think of more flags to pass to Webkit/Blink engines when we `ignoreHTTPSErrors`, other than just `--ignore-certificate-errors`?

File: packages/devtools/tests/launcher.test.ts [hunk 1]
```diff
@@ -331,3 +331,414 @@ test('connect to an existing devtools browser url', async () => {
     expect(vi.mocked(puppeteer.connect))
         .toBeCalledWith(devtoolsOptions)
 })
+
+//asd
```
Comment (christian-bromann) [inline packages/devtools/tests/launcher.test.ts:335]:
```suggestion
```

**Review by christian-bromann — COMMENTED** [review]
> Can you think of more flags to pass to Webkit/Blink engines when we ignoreHTTPSErrors, other than just --ignore-certificate-errors?

Not really. I am not too familiar with all flags.

**Review by christian-bromann — APPROVED** [review]
LGTM 👍

**Comment by nextlevelbeard** [comment]
@christian-bromann how come this was left out of the [changelog](https://github.com/webdriverio/webdriverio/releases/tag/v8.10.1)?

**Comment by christian-bromann** [comment]
I missed to label this PR correctly, let me add it.

**Comment by nextlevelbeard** [comment]
> I missed to label this PR correctly, sorry about that, let me add it manually.

@christian-bromann I see you added it thank you, but there's extra erroneous information unrelated to the feature, has nothing to do with the **browserstack-service**

___
https://github.com/webdriverio/webdriverio/pull/10329 **[browserstack-service]** devtools: Honor acceptInsecureCerts and ignoreHTTPSErrors settings (@ nextlevelbeard)

___

**Comment by christian-bromann** [comment]
Ah sorry , let me update

**Pull request merged**